src/RegistrationMsg.chpl

```diff
@@ -96,6 +99,199 @@ module RegistrationMsg
+    proc attachCategoricalMsg(cmd: string, name: string, 
+                                            st: borrowed SymTab): MsgTuple throws {
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting Categorical components for '%s'".format(cmd, name));
+
+        var repMsg: string;
+                
+        var cats = st.attrib("%s.categories".format(name));
+        var codes = st.attrib("%s.codes".format(name));
+
+        if (cats.startsWith("Error:")) { 
+            var errorMsg = cats;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+        if (codes.startsWith("Error:")) { 
+            var errorMsg = codes;
+            regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+            return new MsgTuple(errorMsg, MsgType.ERROR); 
+        }
+
+        repMsg = "categorical+created %s".format(cats);
+        // Check if the categories is numeric or string, if string add byte size
+        if (isStringAttrib(cats)) {
+            var s = getSegString("%s.categories".format(name), st);
+            repMsg += "+created bytes.size %t".format(s.nBytes);
+        }
+
+        repMsg += "+created %s".format(codes);
+
+        // Optional components of categorical
+        if st.contains("%s.permutation".format(name)) {
+            var perm = st.attrib("%s.permutation".format(name));
+            if (perm.startsWith("Error:")) { 
+                var errorMsg = perm;
+                regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR); 
+            }
+            repMsg += "+created %s".format(perm);
+        }
+        if st.contains("%s.segments".format(name)) {
+            var segs = st.attrib("%s.segments".format(name));
+            if (segs.startsWith("Error:")) { 
+                var errorMsg = segs;
+                regLogger.error(getModuleName(),getRoutineName(),getLineNumber(),errorMsg);
+                return new MsgTuple(errorMsg, MsgType.ERROR); 
+            }
+            repMsg += "+created %s".format(segs);
+        }
+
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+
+    /* 
+    Compile the component parts of a SegArray attach message 
+
+    :arg cmd: calling command 
+    :type cmd: string 
+
+    :arg name: name of SymTab element
+    :type name: string
+
+    :arg st: SymTab to act on
+    :type st: borrowed SymTab 
+
+    :returns: MsgTuple response message
+    */
+    proc attachSegArrayMsg(cmd: string, name: string, st: borrowed SymTab): MsgTuple throws {
+        regLogger.debug(getModuleName(),getRoutineName(),getLineNumber(), 
+                            "%s: Collecting SegArray components for '%s'".format(cmd, name));
+
+        var repMsg: string;
+
+        var segs = st.attrib("%s_segments".format(name));
```

Comment:
i hate having to use `_` as our delimiter for segarray rather than `.`

I get that's to avoid conflicts with registering a categorical with the same name, but I want to be able to use a consistent delimiter for components

Reply:
i don't think it's wrong to tell the user that name is already in use for another kind of object.

Reply:
This might be a good issue for Jim to work on when he gets setup. That should a very straight forward change